community/blog/2023-01 Role-based authentication in Azure API Management made easy in Arcus Scripting v0.8/README.md

```diff
@@ -0,0 +1,63 @@
+# Role-based authentication in Azure API Management made easy in Arcus Scripting v0.8
+Managing roles during JWT validation was tedious and repetitive work. Luckily, the newest Arcus Scripting release changed that.
+
+## Validating JSON web tokens in Azure API Management
+The `validate-jwt` policy in Azure API Management provides the capability to enforce a valid JWT in an incoming HTTP request. One of these validation rules is whether certain roles have been assigned to a service principal. This authorization functionality is very useful to allow/deny certain functionality to users of the application.
+
+An example of such a policy is shown here:
+
+```xml
+<validate-jwt header-name="Authorization" failed-validation-httpcode="401" failed-validation-error-message="Unauthorized. Access token is missing or invalid.">
+    <openid-config url="{{openid-config-url}}" />
+    <issuers>
+        <issuer>{{issuer}}</issuer>
+    </issuers>
+    <required-claims>
+        <claim name="roles" match="any">
+            <value>DummyRole</value>
+        </claim>
+    </required-claims>
+</validate-jwt>
+```
+
+Management of service principal roles to certain Azure Directory Applications is rather tedious. One has to look up the role assignments of an Azure Active Directory application to find out if the service principal has the correct access. Moreover, in certain scenarios, one has to wait a couple of seconds before a role assignment is available for use. All this adds to the problem of managing a service principal for validating JWTs in Azure API Management.
```

Comment:
Here we're talking about a certain problem, but what is the solution ?

Reply:
Our PowerShell module 😄

Reply:
Ok, but I meant that this is not obvious from the text :)

Reply:
But, we have **In a single Arcus Scripting release, we have fully fixed the problem of role assignments for a service principal**, right? In the next paragraph?

Reply:
Maybe we should consider moving that one sentence from that paragraph to this paragraph ?

Reply:
Done in https://github.com/arcus-azure/arcus/pull/290/commits/e770ba5cdbda61bc7ab54ddb305ab9e99ad3729b